src/config.h

```diff
@@ -232,13 +232,21 @@ struct Config {
   };
   Index index;
 
+  // List of folders opened in current workspace.
+  std::vector<std::string> workspaceFolders;
```

Comment:
This probably shouldn't be in config, can you store it as a variable in the base message handler class?

Reply:
Hmmm..
Don't know how to achieve this. Okay, let me put forth my understanding of LSP.
- There are only two ways to receive config options from clients.
    1. Through initialization options - stored in `struct Config`.
    2. Through --init command line option - can be stored in `strut MessageHandler` or in one of its fields.
- Clients don't (can't) send extra options through queries.

Please, correct me if I'm wrong regarding these assumptions.

Coming to the change you suggested, did you mean `struct MessageHandler` or `struct BaseMessageHandler`.
In either case please elaborate on how to achieve this. Thanks.

Reply:
In `MessageHandler`, ie, where `db`, `project` and the like are defined (src/message_handler.h line 85ish). You can declare a `std::unique_ptr<std::vector<std::string>> workspaceFolders` there and append to it inside of init, ie, so that every derived MessageHandler type will use the same `std::vector` instance.

Reply:
Let me know if you don't feel like doing that, a TODO is good enough.

Reply:
Thanks, for clarification. I've made the changes you suggested. Check if that is what you intended.
Please feel free to suggest further changes.